test/test.js

```diff
@@ -154,6 +169,39 @@ describe('DOM with opened saved basket', function () {
 
         chai.assert.equal(afterCartProductCount, initialCartProductCount - 1);
     })
+
+    it('should have correct save cart button text', function () {
+        const btnText = document.querySelector('.smart-basket-save-button').innerText
+        chai.assert.equal(btnText, 'Save cart in "Smart Baskets"')
+    })
+
+    it('should call alert with correct message if cart was saved', function () {
+        let alertSpy = sinon.spy(window, 'alert')
+        saveCurrentCart()
+        chai.assert.equal(alertSpy.getCall(0).args[0], 'Cart "KETO-CHILLI" is now stored in "Smart Baskets"')
+    })
+})
+
+describe('DOM with opened saved basket that is already stored', function () {
+    function saveCurrentCart() {
+        document.querySelector('.smart-basket-save-button').click();
+    }
+
+    beforeEach(function () {
+        setupMockCarts();
+        setupDOM('test/rimi-cart-saved-cart-opened.html')
+    })
+
+    it('should have correct update cart button text', function () {
+        const btnText = document.querySelector('.smart-basket-save-button').innerText
+        chai.assert.equal(btnText, 'Update cart in "Smart Baskets"')
+    })
+
+    it('should call alert with correct message if cart was updated', function () {
+        let alertSpy = sinon.spy(window, 'alert')
```

Comment:
Checking whether a method is called means that this test knows too much about the implementation. If we change from `window.alert` to anything else then the test will break immediately.

It would be best to test the behavior instead - "should create visible element containing `cart has been updated` text".
If you are willing to make the `window.alert` change, I suggest you write that ^ test first (make sure it fails) and only then implement the changes